server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileFactory.java

```diff
@@ -171,6 +172,37 @@ private void setDefault(DbSession session, QualityProfileDto profile) {
     db.qualityProfileDao().update(session, profile.setDefault(true));
   }
 
+  public QualityProfileDto find(QProfileRef ref) {
+    try (DbSession dbSession = db.openSession(false)) {
+      return find(ref, dbSession);
+    }
+  }
+
+  public QualityProfileDto find(QProfileRef ref, DbSession dbSession) {
```

Comment:
I do not agree. I prefer keeping the same order of parameters when method is overloaded: `find(QProfileRef)` and `find(QProfileRef, DbSession)`